lib/tasks/data/text_similarity_data.rake

```diff
@@ -157,10 +158,21 @@ namespace :check do
         puts 'Please provide the AWS credentials.'
         exit 1
       end
+      body = data.collect{ |row| row.to_json }.join("\n")
+      key = "text-similarity-data-export-#{Time.now.strftime('%Y-%m-%d')}.json.zip"
+      # Step 1: Compress data in memory (create an in-memory IO stream)
+      io = StringIO.new
+      gz = Zlib::GzipWriter.new(io)
+      # Write the data into the Gzip stream
+      gz.write(body)
+      gz.close  # Close the stream to finalize compression
+      compressed_data = io.string  # Get the compressed data as a string
+      # Step 2: Upload the compressed file to AWS S3
       response = s3_client.put_object(
         bucket: s3_bucket_name,
-        key: "text-similarity-data-export-#{Time.now.strftime('%Y-%m-%d')}.json",
-        body: JSON.pretty_generate(data)
+        key: key,
+        body: compressed_data,
+        content_type: 'application/zip'
```

Comment:
Sawy, are you sure about this `content_type`? I was expecting a mix of `content_type` and `content_encoding` (which, again, should be Gzip, not Zip):

```
Content-Type: application/json
Content-Encoding: gzip
```

Did you try that?

Reply:
Yes Caio I tried different cases and only work for `content_type: application/gzip` not mix with `content_encoding`